core/src/main/java/org/apache/calcite/rel/rules/SortRemoveRedundantRule.java

```diff
@@ -96,19 +109,33 @@ protected SortRemoveRedundantRule(final SortRemoveRedundantRule.Config config) {
     }
   }
 
-  private Optional<Double> getSortInputSpecificMaxRowCount(Sort sort) {
-    // If the sort is pure limit, the specific max row count is limit's fetch.
-    if (RelOptUtil.isPureLimit(sort)) {
-      final double limit =
-          sort.fetch instanceof RexLiteral ? RexLiteral.intValue(sort.fetch) : -1D;
-      return Optional.of(limit);
+  private Optional<Integer> getSortInputSpecificMaxRowCount(Sort sort) {
```

Comment:
`getSortInputSpecificMaxRowCount` and `getMaxRowCountForLimitAccordingToIsOrder` can be merged because `if` branch can be simplified.